### PR TITLE
Feature/1806 link experiences to users/applicants

### PR DESCRIPTION
**Pull request by JamesHuf**
Addresses #1806.

Added Applicant graphql type which contains the subset of User fields which would be relevant to an interested manager.
Also changed the user seeder to generate experiences.

### CODE REVIEW

**Comment by petertgiles** [comment]
Wow, this is pretty impressive, James.  The implementation looks good to me.  I have a few questions about the design but these might be for you or @tristan-orourke .

- Should there be a query for Applicants implemented here?
- I'm not sure the Experience interface is very useful right now.  I'm guessing it has `details` since that's a field that's common to all experience models but I think what we really want is some kind of `displayName` that can be shown in a list or a chip or something?
- Should the experience models have `@belongsTo` a User or an Applicant?

**Comment by petertgiles** [comment]
In case it's useful, here's a graphql query:
```
query {
  me {
    email
    sub
    experiences {
      id
      details
    }
    awardExperiences {
      title
      issuedBy
      awardedDate
      awardedTo
      awardedScope
      details
    }
    communityExperiences {
      title
      organization
      project
      startDate
      endDate
      details
    }
    educationExperiences {
      institution
      areaOfStudy
      thesisTitle
      startDate
      endDate
      type
      status
      details
    }
    personalExperiences {
      title
      description
      startDate
      endDate
      details
    }
    workExperiences {
      organization
      division
      startDate
      endDate
      details
    }
  }
}
```

**Comment by JamesHuf** [comment]
I think the main benefit of the Experience interface is the ability to query for every experience type in a single array using something like:
`query {
  me {
    email
    sub
    experiences {
      id
      details
        ... on WorkExperience {
          role
          organization
        }
        ... on PersonalExperience {
          description
        }
    }
  }
}`

**Comment by petertgiles** [comment]
> the ability to query for every experience type in a single array using something like

Oh, I didn't know about that.  Very cool!

**Comment by tristan-orourke** [comment]
Like @JamesHuf showed, its already worth having the Experience interface. However, a `displayName` field is a very interesting idea @petertgiles! Each type of experience has a different way of generating its display name, as seen in #1925. Constructing it on the api side might save us from doing it in multiple places on the frontend. Think its worth making an issue for?

**Comment by JamesHuf** [comment]
I've changed experiences to reference Applicants instead of Users and have added queries for applicants. Applicants can currently be queried by anyone because I'm not sure how we plan on determining access to them.

**Review by petertgiles — CHANGES_REQUESTED** [review]
This is looking great!  I don't mind that anyone can query the applicants right now.  At least the hooks are ready for when we want to formalize the roles and policies.  I'm going to put in an issue for it though.

It looks like the seeder for fake data is not populating `details` on `award_experiences` and `education_experiences`.  Could you please double-check those?

File: api/app/Policies/UserPolicy.php [hunk 1]
```diff
@@ -114,4 +114,28 @@ public function forceDelete(User $user, User $model)
...
+    /*******************  APPLICANT QUERIES  *******************/
+    /**
+     * Determine whether the user can view any models.
+     *
+     * @param  \App\Models\User|null  $user
+     * @return \Illuminate\Auth\Access\Response|bool
+     */
+    public function viewAnyApplicants(User $user = null)
+    {
+        return true;
```
Comment (tristan-orourke) [inline api/app/Policies/UserPolicy.php:127]: Should check that user is an admin

File: api/app/Policies/UserPolicy.php [hunk 2]
```diff
@@ -114,4 +114,28 @@ public function forceDelete(User $user, User $model)
...
+    /**
+     * Determine whether the user can view the model.
+     *
+     * @param  \App\Models\User|null  $user
+     * @param  \App\Models\User  $model
+     * @return \Illuminate\Auth\Access\Response|bool
+     */
+    public function viewApplicant(User $user = null, User $model)
+    {
+        return true;
```
Comment (tristan-orourke) [inline api/app/Policies/UserPolicy.php:139]: Should check that user is an admin, OR user and model have the same id.

**Review by tristan-orourke — CHANGES_REQUESTED** [review]
Thanks for the prompt updates. For now, lets use the same logic on the viewApplicant and viewAnyApplicant methods as we already have on viewUser and viewAnyUser. We can make them less restrictive when we have actual manager roles.

**Review by petertgiles — APPROVED** [review]

**Comment by brindasasi** [comment]
@JamesHuf as discussed in the scrum , you could go ahead and merge this PR

**Pull request merged**